tests/data/Q748282-en/output.html

```diff
@@ -1,93 +1,18 @@
-  </li>
-  <li>
-    Northern Demirji (Ukrainian: <span lang="uk">Північний Демірджі</span>, Russian: <span lang="ru">Северный Демирджи</span>, Crimean Tatar: <span lang="crh">Şimaliy Demirci</span>) 1,356 m in the Demirci Yayla;
-  </li>
-  <li>
-    Ai-Petri (Ukrainian: <span lang="uk">Ай-Петрі</span>, Russian: <span lang="ru">Ай-Петри</span>, Crimean Tatar: <span lang="crh">Ay Petri</span>) 1,234 m in the Ay Petri Yaylası.
-  </li>
-</ul>
-<h2>
-  Passes and rivers
-</h2>
-<p>
-  The passes over the Crimean Mountains are: (from east to west)
-</p>
-<ul>
-  <li>
-    Angarskyi Pass (752m) near Perevalne, on a road from Alushta to Simferopol
-  </li>
-  <li>
-    Baydar Gate (503m) near Foros, connecting Baydar Valley and the sea coast
-  </li>
-  <li>
-    Laspi Pass (350m) near Cape Aya, on a road from Yalta to Sevastopol.
-  </li>
-</ul>
-<p>
-  Rivers of the Crimean Mountains include the Alma River, Chernaya River, and Salhir River on the northern slope and Uchan-su River on the southern slope which forms the Uchan-su waterfall, and the highest waterfall in Crimea.
-</p>
-<h2>
-  History
-</h2>
-<p>
-  Archaeologists have found the earliest anatomically modern humans in Europe in the Crimean Mountains' Buran-Kaya caves. The fossils are 32,000 years old, with the artifacts linked to the Gravettian culture. The fossils have cut marks suggesting a post-mortem defleshing ritual.
-</p>
\ No newline at end of file
+<p>The <b>Crimean Mountains</b> (Crimean Tatar: <span lang="crh">Qırım dağları</span>; Ukrainian: <span lang="uk">Кримські гори</span>; Russian: <span lang="ru">Крымские горы</span>; Turkish: <i lang="tr">Yayla Dağları</i>) or <b>Yayla Mountains</b> are a range of mountains running parallel to the south-eastern coast of Crimea, between about 8–13 kilometers (5–8 miles) from the sea. Toward the west, the mountains drop steeply to the Black Sea, and to the east, they change slowly into a steppe landscape.</p><p>The Crimean Mountains consist of three subranges. The highest is the Main Range, which is subdivided into several yaylas or mountain plateaus (<i>yayla</i> or <i>yaylak</i> is Turkic for "alpine meadow"). They are:</p><ul><li>Baydar yayla</li>
+<li>Ai-Petri yayla</li>
+<li>Yalta yayla</li>
+<li>Nikita yayla</li>
+<li>Hurzuf yayla</li>
+<li>Babugan yayla</li>
+<li>Chatyr-Dag yayla</li>
+<li>Dologorukovskaya (Subatkan) yayla</li>
+<li>Demirci yayla</li>
+<li>Qarabiy yayla</li></ul><h2>Highest peaks</h2><p>The Crimea's highest peak is the Roman-Kosh (Ukrainian: <span lang="uk">Роман-Кош</span>; Russian: <span lang="ru">Роман-Кош</span>, Crimean Tatar: <span lang="crh">Roman Qoş</span>) on the Babugan Yayla at 1,545 metres (5,069&nbsp;ft). Other important peaks over 1,200 metres include:</p><ul><li>Demir-Kapu (Ukrainian: <span lang="uk">Демір-Капу</span>, Russian: <span lang="ru">Демир-Капу</span>, Crimean Tatar: <span lang="crh">Demir Qapı</span>) 1,540 m in the Babugan Yayla;</li>
```

Comment:
Would it save a bit more space if the nbsp were encoded directly as ` `, instead of `&nbsp;`?

Reply:
It would, but we don't control that part of the writing. `html5ever` [converts the literal to the escaped version](https://github.com/servo/html5ever/blob/e69b05c849031d6a2837d7d86372ff14b3f4080e/html5ever/src/serialize/mod.rs#L108), I assume because it is [part of the serialization spec](https://html.spec.whatwg.org/multipage/parsing.html#escapingString).
It's possible to write another `Serializer` like the pretty-printer that minifies instead, but I haven't figured out the whitespace collapsing rules enough to write one. There aren't any crates that implement an `html5ever::Serializer` minifier, so adding an external minifier would need to re-parse the html.